src/telliot_feeds/reporters/customized/backup_reporter.py

```diff
@@ -0,0 +1,184 @@
+        current_timestamp = self.current_time()
+        round_id = chainlink_latest_round_data.roundId
+        updated_at = chainlink_latest_round_data.updatedAt
+        answer = chainlink_latest_round_data.answer
+        if round_id == 0:
+            logger.info(f"chainlink is broken. round id: {round_id}")
+            return True
+        if updated_at == 0 or int(updated_at) > current_timestamp:
+            logger.info(f"chainlink is broken. updated at: {updated_at}")
+            return True
+        if answer == 0:
+            logger.info(f"chainlink is broken. answer: {answer}")
+            return True
+        return False
+
+    async def get_tellor_latest_data(self) -> Optional[GetDataBefore]:
+        if self.datafeed is None:
+            logger.debug(f"no datafeed set: {self.datafeed}")
+            return None
+        data, status = await self.oracle.read("getDataBefore", self.datafeed.query.query_id, self.current_time())
+        if not status.ok:
+            logger.warning(f"error getting tellor data: {status.e}")
+            return None
+        return GetDataBefore(*data)
+
+    async def conditions_met(self) -> bool:
+        """Trigger methods to check conditions if reporting spot is necessary"""
+        logger.info("checking conditions and reporting if necessary")
+        tellor_data = await self.get_tellor_latest_data()
+
+        current_time = self.current_time()
+        time_passed_since_tellor_report = current_time - tellor_data.timestampRetrieved if tellor_data else current_time
+        if (
+            tellor_data is None  # if there is tellor data check chainlink feeds
+            or tellor_data.retrieved is False
+            # if tellor data is old check chainlink feeds
+            or time_passed_since_tellor_report > self.chainlink_is_frozen_timeout
+        ):
+            chainlink_latest_round_data = self.get_chainlink_latest_round_data()
+
+            if chainlink_latest_round_data is None:
+                return True
+            if self.chainlink_is_broken(chainlink_latest_round_data):
+                return True
+            if self.chainlink_is_frozen(chainlink_latest_round_data):
+                return True
+            chainlink_previous_round_data = self.get_chainlink_previous_round_data(chainlink_latest_round_data.roundId)
+
+            if chainlink_previous_round_data is None:
+                return True
+
+            if self.chainlink_price_change_above_max(chainlink_latest_round_data, chainlink_previous_round_data):
+                return True
+            logger.info(f"chainLink {self.datafeed.query.descriptor} data is recent enough")  # type: ignore
+            return False
+        else:
+            logger.info(f"tellor {self.datafeed.query.descriptor} data is recent enough")  # type: ignore
+            return False
+
+    async def report(self, report_count: Optional[int] = None) -> None:
```

Comment:
A bit of duplicate code in this overwrite of `report()` in the `IntervalReporter` class. You could rewrite the original to take a list of other optional conditions to check, or a list of optional callable functions to check. Make a new report function variation `foo()` in this file that contains a call to the updated `report` method, passing the callable `conditions_met` to it. Then in the `liquity()` command, call your new report function `foo()` instead of the original `report()` one.

Reply:
This looks ok to me since I want to call those methods still with just an additional custom method.